### PR TITLE
[VideoDatabase] Improved speed of CVideoDatabase::GetEpisodeId: Only …

### PR DESCRIPTION
**Pull request by xLAva**
…get basic episode information directly from the "episode" table instead of the "episode_view"

## Description
Only very basic information is needed during the library update check.
This change reduces the fetched data and it also only fetches information from one table and does not need the slower table view.

## Motivation and Context
When using a MySQL database back-end on a slow remote server (like a Raspberry Pi), the library update check on existing episodes can be very slow. 
Especially with TV shows with many seasons and episodes.
I did not exactly measure the improvement, but for certain shows the update time was reduced from multiple minutes to seconds on my setup.

## How Has This Been Tested?
Tested on my running setup for a year:
* MySQL server on Raspberry Pi (Ubuntu 16.04 / Ubuntu 18.04)
* Kodi clients on multiple Linux workstations (Ubuntu 16.04 / Ubuntu 18.04)

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/video/VideoDatabase.cpp [hunk 1]
```diff
@@ -3991,12 +4012,12 @@ CVideoInfoTag CVideoDatabase::GetDetailsForTvShow(const dbiplus::sql_record* con
   return details;
 }
 
-CVideoInfoTag CVideoDatabase::GetDetailsForEpisode(std::unique_ptr<Dataset> &pDS, int getDetails /* = VideoDbDetailsNone */)
+CVideoInfoTag CVideoDatabase::GetBasicDetailsForEpisode(std::unique_ptr<Dataset> &pDS)
```
Comment (Razzeee) [inline xbmc/video/VideoDatabase.cpp]: I don't think you wanted to remove/rename the old methods?

Comment (Rechi) [reply xbmc/video/VideoDatabase.cpp]: read the full diff

File: xbmc/video/VideoDatabase.cpp [hunk 2]
```diff
@@ -2073,6 +2073,27 @@ bool CVideoDatabase::GetSeasonInfo(int idSeason, CVideoInfoTag& details, bool al
   return false;
 }
 
+bool CVideoDatabase::GetEpisodeBasicInfo(const std::string& strFilenameAndPath, CVideoInfoTag& details, int idEpisode /* = -1 */)
+{
+  try
+  {
+    if (idEpisode < 0)
+      idEpisode = GetEpisodeId(strFilenameAndPath);
+    if (idEpisode < 0) return false;
```
Comment (notspiff) [inline xbmc/video/VideoDatabase.cpp]: please no single line if's. think of the poor debugger.

File: xbmc/video/VideoDatabase.cpp [hunk 3]
```diff
@@ -4010,6 +4031,30 @@ CVideoInfoTag CVideoDatabase::GetDetailsForEpisode(const dbiplus::sql_record* co
...
+CVideoInfoTag CVideoDatabase::GetDetailsForEpisode(std::unique_ptr<Dataset> &pDS, int getDetails /* = VideoDbDetailsNone */)
+{
+  return GetDetailsForEpisode(pDS->get_sql_record(), getDetails);
+}
+
+CVideoInfoTag CVideoDatabase::GetDetailsForEpisode(const dbiplus::sql_record* const record, int getDetails /* = VideoDbDetailsNone */)
+{
+  CVideoInfoTag details;
+
+  if (record == NULL)
```
Comment (notspiff) [inline xbmc/video/VideoDatabase.cpp]: nullptr.

File: xbmc/video/VideoDatabase.cpp [hunk 4]
```diff
@@ -4010,6 +4031,30 @@ CVideoInfoTag CVideoDatabase::GetDetailsForEpisode(const dbiplus::sql_record* co
...
+    return details;
+
+  details = GetBasicDetailsForEpisode(record);
+  
+  DWORD time = XbmcThreads::SystemClockMillis();
```
Comment (notspiff) [inline xbmc/video/VideoDatabase.cpp]: please use a real type (unsigned int?) not this uppercase thing.

**Review by notspiff — CHANGES_REQUESTED** [review]
just a few small things, otherwise it looks good to me.

**Comment by xLAva** [comment]
@notspiff - I implemented your small change requests.

**Comment by notspiff** [comment]
thx

**Pull request merged**